core/DataAccess/Model.php

```diff
@@ -1002,11 +1020,84 @@ private function getRemoveInvalidationsIdSitesClause($idSite)
         return "idsite IN ($idSitesStr) AND";
     }
 
-    public function releaseInProgressInvalidation($idinvalidation)
+    /**
+     * Releases in progress invalidations for the given ids
+     *
+     * To avoid duplicate invalidations in the database, the method is also meant to prevent having duplicates after a reset
+     * Therefor below code will check if any of the invalidations to be reset should be removed instead
+     * An invalidation can be savely removed
+     *  - if there exists another queued invalidation with the same parameters
+     *  - if there is another running invalidation, that had been started after the current one was invalidated
+     * Otherwise the invalidation will be reset
+     *
+     * @param array $idinvalidations
+     * @return int
+     * @throws \Zend_Db_Statement_Exception
+     */
+    public function releaseInProgressInvalidations(array $idinvalidations): int
     {
+        $idinvalidations = array_map('intval', $idinvalidations);
         $table = Common::prefixTable('archive_invalidations');
-        $sql = "UPDATE $table SET status = " . ArchiveInvalidator::INVALIDATION_STATUS_QUEUED . ", ts_started = NULL, processing_host = NULL, process_id = NULL WHERE idinvalidation = ?";
-        Db::query($sql, [$idinvalidation]);
+        $changedCount = 0;
+
+        $sql = "SELECT * FROM $table WHERE idinvalidation IN (" . implode(',', $idinvalidations) . ")";
+        $invalidations = Db::fetchAll($sql);
+
+        // Check invalidations one by one, to ensure we safely remove invalidations in cases where two identical ones are requested to reset
+        foreach ($invalidations as $invalidation) {
+            // Look for other indentical invalidations that are either not started or started after the current one had been invalidated
+            $query = "SELECT COUNT(*) FROM $table WHERE name = ? AND idsite = ? AND date1 = ? AND date2 = ? AND period = ? AND "
+                   . "(status = ? OR (status = ? AND ts_started > ?)) AND idinvalidation != " . $invalidation['idinvalidation'];
```

Comment:
```suggestion
                   . "(status = ? OR (status = ? AND ts_started > ?)) AND idinvalidation != ?";
```

Just to be consistent?